### PR TITLE
appgate-sdp: 5.1.2 -> 5.3.2

### PR DESCRIPTION
**Pull request by ymatsiuk**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by ymatsiuk** [comment]
Result of `nixpkgs-review pr 110347` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>appgate-sdp</li>
  </ul>
</details>

File: pkgs/applications/networking/appgate-sdp/default.nix [hunk 1]
```diff
@@ -92,11 +96,11 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "appgate-sdp";
-  version = "5.1.2";
+  version = "5.3.2";
 
   src = fetchurl {
-    url = "https://bin.appgate-sdp.com/5.1/client/appgate-sdp_${version}_amd64.deb";
-    sha256 = "0v4vfibg1giml3vfz2w7qypqzymvfchi5qm6vfagah2vfbkw7xc2";
+    url = "https://bin.appgate-sdp.com/5.3/client/appgate-sdp_${version}_amd64.deb";
```
Comment (SuperSandro2000) [inline pkgs/applications/networking/appgate-sdp/default.nix:102]: Could we calculate this number?

**Comment by etu** [comment]
@ymatsiuk I just found out that this was packaged through this PR. You're my hero now. I've needed this at work for months but it's so many people that have issues with it that I've just said that it "doesn't work for me". I'll test this on our company "vpn" tomorrow :)

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 110347` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>appgate-sdp</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

appgate-sdp:
<details>

Please consider this feature to be alpha.

A substituteInPlace with an unmatched pattern got detected:
```
substituteStream(): WARNING: pattern '/opt/' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/appgate-resolver.pre'
substituteStream(): WARNING: pattern '/usr/sbin/dnsmasq' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/appgate-resolver.pre'
substituteStream(): WARNING: pattern 'InaccessiblePaths=/mnt /srv /boot /media' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/appgate-resolver.pre'
substituteStream(): WARNING: pattern '/opt/' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/appgate-dumb-resolver.pre'
substituteStream(): WARNING: pattern '/usr/sbin/dnsmasq' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/appgate-dumb-resolver.pre'
substituteStream(): WARNING: pattern 'InaccessiblePaths=/mnt /srv /boot /media' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/appgate-dumb-resolver.pre'
substituteStream(): WARNING: pattern '/usr/sbin/dnsmasq' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgatedriver.service'
substituteStream(): WARNING: pattern 'cat' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgatedriver.service'
substituteStream(): WARNING: pattern 'pkill' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgatedriver.service'
substituteStream(): WARNING: pattern '/usr/sbin/dnsmasq' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-dumb-resolver.service'
substituteStream(): WARNING: pattern 'InaccessiblePaths=/mnt /srv /boot /media' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-dumb-resolver.service'
substituteStream(): WARNING: pattern 'cat' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-dumb-resolver.service'
substituteStream(): WARNING: pattern 'pkill' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-dumb-resolver.service'
substituteStream(): WARNING: pattern 'InaccessiblePaths=/mnt /srv /boot /media' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-resolver.service'
substituteStream(): WARNING: pattern 'cat' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-resolver.service'
substituteStream(): WARNING: pattern 'pkill' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/lib/systemd/system/appgate-resolver.service'
substituteStream(): WARNING: pattern 'service appgate-resolver stop' doesn't match anything in file '/nix/store/yr3zkwws4122bk0l8scdv94hvklhvia6-appgate-sdp-5.3.2/opt/appgate/linux/set_dns'
```

Please check the offending substituteInPlace for typos or changes in source.
</details>

**Comment by ymatsiuk** [comment]
@SuperSandro2000 last time I've been asking this I got [this](https://github.com/NixOS/nixpkgs/pull/108402#issuecomment-756121328). But sure, let me fix this 😄

**Comment by etu** [comment]
@SuperSandro2000 It's not uncommon that we loop over a list of files and do substituteInPlace with things that don't exist in all the files. Mostly because it's cleaner code and nicer. The warnings doesn't matter. It's more important with code that is easier to read. I see no need for @ymatsiuk to change that.

**Comment by SuperSandro2000** [comment]
> The warnings doesn't matter.

In this case the warnings do not matter but I am not sure if and how we want to filter this.

**Comment by ymatsiuk** [comment]
Fixed warning and added shell substring magic for version calculation @SuperSandro2000 😃

File: pkgs/applications/networking/appgate-sdp/default.nix [hunk 2]
```diff
@@ -92,11 +96,11 @@ let
 in
 stdenv.mkDerivation rec {
   pname = "appgate-sdp";
-  version = "5.1.2";
+  version = "5.3.2";
 
   src = fetchurl {
-    url = "https://bin.appgate-sdp.com/5.1/client/appgate-sdp_${version}_amd64.deb";
-    sha256 = "0v4vfibg1giml3vfz2w7qypqzymvfchi5qm6vfagah2vfbkw7xc2";
+    url = "https://bin.appgate-sdp.com/${version:0:3}/client/appgate-sdp_${version}_amd64.deb";
```
Comment (etu) [inline pkgs/applications/networking/appgate-sdp/default.nix:102]: This won't work, it we don't do substrings like this. You probably want the following:
```suggestion
    url = "https://bin.appgate-sdp.com/${lib.versions.majorMinor version}/client/appgate-sdp_${version}_amd64.deb";
```

**Review by etu — CHANGES_REQUESTED** [review]
I have just pulled this in on my system with the fixed download link (as per my comment) through an overlay and rebooted with that as a generation. And I must say, it works, and this will make my life so much easier :smile: 

I'm ready to merge this whenever the link is fixed.

**Review by etu — APPROVED** [review]

**Comment by ymatsiuk** [comment]
@etu Thanks for suggestion. Applied.

PS: I've seen you mentioned `appgate` in IRC but I did some research and found 0 nix expressions. I was going to poke you, to help me test this out. I'm using it on a daily basis for a month already and see no issues. And BTW I've contacted the AUR maintainer (who works for the company owner of appgate) and he confirmed that tray icon doesn't work curresntly 🤷🏻‍♂️

**Pull request merged**